config/initializers/highlight_override.rb

```diff
@@ -0,0 +1,54 @@
+# Monkey patch the generated Bindings for the swagger Highlight model
+
+Api::V0::Bindings::NewHighlight.class_exec do
+  def valid_location_strategies?
+    location_strategies.present? &&
+      location_strategies.none? { |item| item.nil? } &&
+      location_strategies.all?(&:valid?)
+  end
+
+  def location_strategies=(array)
+    @location_strategies = array.map do |item|
+      case item[:type]
+      when 'TextPositionSelector'
+        Api::V0::Bindings::TextPositionSelector.new(item)
+      when 'XpathRangeSelector'
+        Api::V0::Bindings::XpathRangeSelector.new(item)
+      end
+    end
+  end
+
+  alias_method :old_valid?, :valid?
+  def valid?
+    old_valid? && valid_location_strategies?
+  end
+
+  alias_method :old_list_invalid_properties, :list_invalid_properties
+  def list_invalid_properties
+    invalid_properties = old_list_invalid_properties
+
+    if location_strategies.blank?
+      invalid_properties.push('invalid value for "location_strategies", location_strategies cannot be empty.')
+    end
+
+    if location_strategies.any? { |item| item.nil? }
+      invalid_properties.push('invalid strategy detected')
```

Comment:
`location_strategies.any?(&:nil?)` and then maybe more meaning in the error message, e.g. ‘location strategies cannot be null’?  and generally maybe should make these error messages look a little more similar (this one is different than the above’s format, for example)

Reply:
in this case, which is covered by a test, is when a location strategy is passed in that's not one of the two we recognize.  it then gets set to nil (in the setter) which is then handled here (perhaps an empty array element was passed in, or as described here, an invalid foobar one.  

in this case, it returns "invalid strategy detected" without making the code unnecessarily complex.  

that was the idea.